packages/pwa-kit-react-sdk/src/ssr/universal/components/storefront-preview/index.jsx

```diff
@@ -14,14 +14,16 @@ import {detectStorefrontPreview, getClientScript} from './utils'
  *
  * @param {boolean} enabled - flag to turn on/off Storefront Preview feature
  * @param  {function(): string | Promise<string>} getToken - A method that returns the access token for the current user
+ * @param  {Array} getAdditionalSearchParams - A method that returns an array of key/value search params to append when context changes
  */
-export const StorefrontPreview = ({enabled = true, getToken}) => {
+export const StorefrontPreview = ({enabled = true, getToken, getAdditionalSearchParams}) => {
     let isHostTrusted
     useEffect(() => {
         if (enabled && isHostTrusted) {
             window.STOREFRONT_PREVIEW = {
                 ...window.STOREFRONT_PREVIEW,
-                getToken
+                getToken,
+                getAdditionalSearchParams
             }
         }
     }, [enabled, getToken])
```

Comment:
```suggestion
    }, [enabled, getToken, getAdditionalSearchParams])
```
Need to add this to the dependency array to ensure that `window.STOREFRONT_PREVIEW` is updated any time the prop is updated.

Reply:
thanks, updated.